fix: misc doc updates (#1153)
NVIDIA-NeMo/Automodel: docs/guides/llm/nanogpt-pretraining.md

Reviewer: ```suggestion
## Pre-Process the FineWeb Dataset
```

Reviewer: ```suggestion
### Pre-Processing and Tokenization
```

Reviewer: ```suggestion
NeMo AutoModel follows a simple, but powerful flow for training:
```

Reviewer: ```suggestion
### Hugging Face Models using the `from_config` Function
```

Reviewer: delete line ---

Reviewer: ```suggestion
## Pre-Process the FineWeb Dataset
```

Reviewer: ```suggestion
## Understand the NeMo AutoModel Training Workflow
```

Reviewer: ```suggestion
NeMo AutoModel follows a simple but powerful flow for training:
```

Reviewer: ```suggestion
NeMo AutoModel relies on a YAML-driven configuration to build every training component. In particular, the `model._target_` must reference a callable that returns an `nn.Module` (or a compatible Hugging Face model). You can point `_target_` at:
```

Reviewer: ```suggestion
### Quick Introduction to the FineWeb Dataset
```

Reviewer: ```suggestion
2. **Mixed precision**: FSDP2 + `bfloat16` (`dtype: bfloat16` in distributed config) for memory savings.
3. **Sequence packing**: Set `packed_sequence.packed_sequence_size` > 0 to pack variable-length contexts and boost utilization.
4. **Custom datasets**: Implement your own `IterableDataset` or convert existing corpora to the `.bin` format using `tools/nanogpt_data_processor.py` as a template.
5. **BOS alignment**: Set `align_to_bos: true` in the dataset config to ensure sequences start with BOS tokens (requires `bos_token` parameter).
```

Reviewer: ```suggestion
* Periodic checkpoints can be loaded with the `TrainFinetuneRecipeForNextTokenPrediction.load_checkpoint()`.
```

Reviewer: ```suggestion
**How the preprocessor works:** The script streams data iteratively from the Hugging Face Hub (avoiding loading the entire dataset into memory), uses a multiprocessing pipeline with separate reader and writer processes, and parallelizes tokenization across multiple CPU cores using `ProcessPoolExecutor`. This design enables efficient processing of very large datasets while maintaining low memory overhead. By default, it uses the `gpt2` tokenizer, but can support other tokenizers using the `--tokenizer` option.
```

Reviewer: ```suggestion
With this context, let’s define a model using `_target_`, then point the dataset at your preprocessed shards, and finally review the full YAML.
```

Reviewer: ```suggestion
Below is the minimal GPT-2 [implementation](https://github.com/NVIDIA-NeMo/Automodel/blob/main/nemo_automodel/components/models/gpt2.py) used for this NanoGPT-style pretraining flow.
It is a pure-PyTorch model with tied embeddings and standard transformer blocks:
```

Reviewer: ```suggestion
In short, `build_gpt2_model(...)` constructs a compact GPT-2 with configurable depth/width/heads and returns an `nn.Module` that outputs logits over the vocabulary. It’s intentionally lean (no KV-cache or generation helpers), but perfectly suited for forward/backward passes and next-token prediction.
```

Reviewer: ```suggestion
* Defines a GPT-2 model using the `build_gpt2_model` shorthand (easy to scale up).
```

# LLM Pre-Training with NeMo AutoModel

This guide covers **FineWeb** data preparation, **defining** a [NanoGPT‑style](https://github.com/KellerJordan/modded-nanogpt) model, and **launching and monitoring** a NeMo AutoModel pre‑training run.

## Set Up Your Environment

In this guide, we will use an interactive environment to install NeMo AutoModel from Git. You can also install NeMo AutoModel from PyPI or use our bi-monthly Docker container (see the [Installation Guide](../installation.md)).

```bash
# clone / install AutoModel (editable for local hacks)
cd /path/to/workspace/ # specify to your path as needed.
git clone https://github.com/NVIDIA-NeMo/Automodel.git
cd Automodel/
pip install -e ".[all]"    # installs NeMo AutoModel + optional extras
```

> :::note

Reviewer: ```suggestion
:::{note}
```

# LLM Pre-Training with NeMo AutoModel

This guide covers **FineWeb** data preparation, **defining** a [NanoGPT‑style](https://github.com/KellerJordan/modded-nanogpt) model, and **launching and monitoring** a NeMo AutoModel pre‑training run.

> ---

Reviewer: delete line ---

> To run this guide on a single GPU, use the single-GPU command in the **Launch Training** section below and scale down the YAML (for example, reduce `step_scheduler.global_batch_size` / `local_batch_size`, and shrink the model via `model.n_layer` / `model.n_embd` / `model.n_head`). For more launch patterns, see [Run on Your Local Workstation](../../launcher/local-workstation.md).

Reviewer: ```suggestion
:::{tip}
To run this guide on a single GPU, use the single-GPU command in the **Launch Training** section below and scale down the YAML (for example, reduce `step_scheduler.global_batch_size` / `local_batch_size`, and shrink the model using `model.n_layer` / `model.n_embd` / `model.n_head`). For more launch patterns, see [Run on Your Local Workstation](../../launcher/local-workstation.md).
```

:::

> ## Pre-Process the FineWeb Dataset

Reviewer: ```suggestion
## Preprocess the FineWeb Dataset
```

To train on more than 2B tokens from FineWeb, see [pretraining.md](pretraining.md) which uses Megatron Core's sharded dataset format without file size constraints.
:::

> ### Pre-Processing and Tokenization

Reviewer: ```suggestion
### Preprocessing and Tokenization
```

3. Use `--num-workers` to control parallelization.
4. Specify `--output-dir` to change the output location.

## Understand the NeMo AutoModel Training Workflow

NeMo AutoModel follows a simple but powerful flow for training:

> 1. A Python recipe script (for example, [`examples/llm_pretrain/pretrain.py`](https://github.com/NVIDIA-NeMo/Automodel/blob/main/examples/llm_pretrain/pretrain.py)) serves as the entry point that wires up all training components based on a YAML configuration file. Any configuration option can be overridden via CLI arguments (e.g., `--model.name abc`).

Reviewer: ```suggestion
1. A Python recipe script (for example, [`examples/llm_pretrain/pretrain.py`](https://github.com/NVIDIA-NeMo/Automodel/blob/main/examples/llm_pretrain/pretrain.py)) serves as the entry point that wires up all training components based on a YAML configuration file. Any configuration option can be overridden using CLI arguments (e.g., `--model.name abc`).
```

- Any nested mapping that includes `_target_` (e.g., `config:`) is instantiated first and its result is passed upward. This is how the Hugging Face `from_config` pattern works.

> - You can keep using the same training recipe (optimizer, data, distributed settings); only the `model:` block changes.

Reviewer: ```suggestion
:::{note}
- The `model._target_` may reference an import path or a local Python file using the `path.py:object` form.
- Any nested mapping that includes `_target_` (e.g., `config:`) is instantiated first and its result is passed upward. This is how the Hugging Face `from_config` pattern works.
- You can keep using the same training recipe (optimizer, data, distributed settings); only the `model:` block changes.
:::
```
